design-proposals/volume-migration/volume-migration.md

```diff
@@ -0,0 +1,315 @@
+# Overview
+This design proposal extends KubeVirt APIs to enable the storage migration between PVCs.
+
+## Motivation
+In some circumstances, the guest owner may want to have the option of switching the storage provider or class for a group of volumes used by an active VM. The VM must currently be turned off and recreated for this operation causing service disruptions and downtime.
+
+
+## Goals
+Design and develop an API to handle storage migration between PVCs.
+
+ ## Non  Goals
+*  This features is not directly aimed for live migrating the VM using RWO volumes. Hence, the API covers and refers only to the volume migration.
+*  This design doesn't cover migration between storage classes, but rather implements a basic APIs that can be used by overlying tooling
+*  This design doesn't include the creation of the destination PVCs, it expects the destination PVCs to exist.
+
+ ## Definition of Users
+This feature addresses for all KubeVirt users who want to move the PVCs assigned to a VM without workload interruptions.
+
+## User Stories
+
+  1. As a VM owner, the storage class of volumes used by a running VM is deprecated, and the volumes need to be moved to another storage class
+  2. As a VM owner, I want to use higher performance storage class that is now available in the cluster
+  3. As a VM owner, I see my non-resizable storage device filling up and I want to move my VM to a larger storage device without interruption in order to give it more storage
+  4. As an infra admin, I need to replace the storage array due to a hardware refresh and I want to do this without stopping the VMs in order to make this infra operation transparent to VM owners.
+
+## Repos
+
+  - [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+This proposal introduces a new CRD called `VolumeMigration`
+
+```golang
+type VolumeMigration struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec VolumeMigrationSpec `json:"spec"`
+	Status *VolumeMigrationStatus `json:"status,omitempty"`
```

Comment:
When looking at other KubeVirt CRs they do not use `Status` as pointer. E.g.:

```go
type VirtualMachineInstanceMigration struct {
	...
	Status            VirtualMachineInstanceMigrationStatus `json:"status,omitempty"`
}
```

Shall we do the same here?

```suggestion
	Status VolumeMigrationStatus `json:"status,omitempty"`
```

Reply:
In general if a field is omitempty you should make it a pointer. But with the status I agree it should not be since everyone else doesn't have it a pointer, and having to check if the status is nil all over the place can be annoying.

Reply:
Done